api/src/main/java/org/hawkular/apm/api/utils/NodeUtil.java

```diff
@@ -73,6 +89,119 @@ public static boolean isOriginalURI(Node node, String uri) {
     }
 
     /**
+     * This method checks whether a URI template has been defined. If so, then it will
+     * extract the relevant properties from the URI and rewrite the URI to be the template.
+     * This results in the URI being a stable/consistent value (for aggregation purposes)
+     * while retaining the parameters from the actual URI as properties.
+     *
+     * @param node The node containing the current URI and optional template
+     * @return Whether the node was processed
+     */
+    public static boolean rewriteURI(Node node) {
+        boolean processed = false;
+
+        // Check if URI and template has been defined
+        if (node.getUri() != null && node.hasProperty(Constants.PROP_HTTP_URL_TEMPLATE)) {
+            List<String> queryParameters = new ArrayList<String>();
+            String template = node.getProperties(Constants.PROP_HTTP_URL_TEMPLATE).iterator().next().getValue();
+
+            // If template contains a query string component, then separate the details
+            if (template != null && template.indexOf('?') != -1) {
+                int index = template.indexOf('?');
+                String queryString = template.substring(index + 1);
+
+                template = template.substring(0, index);
+
+                StringTokenizer st = new StringTokenizer(queryString, "&");
+
+                while (st.hasMoreTokens()) {
+                    String token = st.nextToken();
+
+                    if (token.charAt(0) == '{' && token.charAt(token.length() - 1) == '}') {
+                        queryParameters.add(token.substring(1, token.length() - 1));
+                    } else {
+                        log.severe("Expecting query parameter template, e.g. {name}, but got '" + token + "'");
+                    }
+                }
+            }
+
+            StringTokenizer uriTokens = new StringTokenizer(node.getUri(), "/");
+            StringTokenizer templateTokens = new StringTokenizer(template, "/");
+
+            if (uriTokens.countTokens() == templateTokens.countTokens()) {
```

Comment:
Is this possible in jax-rs? Either way, would it be ok if I raise a separate jira to check/implement, as I would like to get this in asap for some demos.